sys/shell/shell.c

```diff
@@ -271,7 +310,13 @@ static int readline(char *buf, size_t size)
 #endif
         }
         else {
-            *line_buf_ptr++ = c;
+            /* Always consume characters, but do not not always store them */
+            if ((size_t)curr_pos < size - 1) {
+                buf[curr_pos++] = c;
+            }
+            else {
+                length_exceeded = true;
+            }
```

Comment:
missing `return` with cases for `length_exceeded` or not?

Reply:
No, the function only returns on enter or ctrl-c or EOF. Otherwise it makes it very hard to write robust scripts that interact with the shell as the remaining characters after the length is exceeded would go on to form another line.

Reply:
ah okay